bindings/Sofa/src/SofaPython3/Sofa/Types/Binding_BoundingBox.cpp

```diff
@@ -27,16 +27,20 @@ along with sofaqtquick. If not, see <http://www.gnu.org/licenses/>.
 
 #include "Binding_BoundingBox.h"
 
+#include <SofaPython3/Sofa/Core/Binding_BaseData.h>
+
 #include <sofa/core/objectmodel/Data.h>
-#include <SofaPython3/DataHelper.h>
+#include <sofa/defaulttype/BoundingBox.h>
 #include <SofaPython3/PythonFactory.h>
 
+using sofa::defaulttype::BoundingBox;
+
 namespace sofapython3 {
 
 void moduleAddBoundingBox(py::module& m)
 {
-    py::class_<BoundingBoxData, BaseData,
-            raw_ptr<BoundingBoxData> > bbox(m, "BoundingBox");
+    py::class_<sofa::Data<BoundingBox>, sofa::core::objectmodel::BaseData,
```

Comment:
About that, I'm not sure I'm for Data<XX> instead of XXData or DataXX. The drawback I see with Data<XX> is that there is no way to detect that the AbstractTypeInfo and Name/name() are there or not. But this is for a general discussion at sofa-dev meeting and outside the scope of the PR.

Reply:
I'm not sure I understand what you mean here.

Using an empty class `BoundingBoxData` here forced us to do the `reinterpret_cast` later on, which cause a segmentation faults half the time (weird right? I guess this is what you get for using dangerous casts).

Reply:
I don't understand why using an inherited type would be more/less safe. Actually this is how DataFileName is implemented does it means we cannot bind it ?

Reply:
You are casting an object to a class of another type.
``
`Node::BoundingBox` has been instanciated from the type `sofa::Data<BoundingBox>`, not from `BoundingBoxData`, even if you make `BoundingBoxData` inherit `sofa::Data<BoundingBox>`.

Forcing it to be a `BoundingBoxData`, which it is not, is undefined behavior. My guess would be that the two structures did not align in memory (BoundingBoxData was probably having some an empty byte of padding which is common in empty struct). Hence the `reinterpret_cast` segmentation fault.